### PR TITLE
Added missing unit tests for manager and adapters

### PR DESCRIPTION
**Pull request by Ankit152**
### Check List Before Merging

- [x] This PR passes all pre-commit hook validations.
- [ ] This PR is fully tested and regression tests are included.
- [ ] Relevant documentation has been updated.

### CODE REVIEW

File: cmd/package-operator-manager/bootstrap/apihelper_test.go [hunk 1]
```diff
@@ -87,3 +87,201 @@ func TestIsPKOAvailable(t *testing.T) {
...
+				depl.Status.UpdatedReplicas = depl.Status.AvailableReplicas
+			}).
+			Return(nil)
+
+		isPKODeploymentAvailable, err := isPKODeploymentAvailable(
+			context.Background(), c, "")
+		require.NoError(t, err)
+		assert.True(t, isPKODeploymentAvailable)
+	})
+}
```
Comment (kostola) [inline cmd/package-operator-manager/bootstrap/apihelper_test.go:287]: I might be wrong but this to me looks identical to https://github.com/package-operator/package-operator/blob/7186448b9e86a9920496b3d3155c8e8e117e5669/cmd/package-operator-manager/bootstrap/apihelper_test.go#L20-L89

File: cmd/package-operator-manager/components/components_test.go [hunk 2]
```diff
@@ -34,3 +34,24 @@ func TestUncachedClient(t *testing.T) {
 	require.EqualError(t, err,
 		"unable to set up uncached client: must provide non-nil rest.Config to client.New")
 }
+
+func TestProvideRestConfig(t *testing.T) {
+	t.Parallel()
+
+	_, err := ProvideRestConfig()
+	require.Error(t, err)
```
Comment (kostola) [inline cmd/package-operator-manager/components/components_test.go:42]: What error are we expecting here?

File: cmd/package-operator-manager/components/components_test.go [hunk 3]
```diff
@@ -34,3 +34,24 @@ func TestUncachedClient(t *testing.T) {
...
+
+	_, err := ProvideRESTMapper()
+	require.NoError(t, err)
+}
+
+func TestProvideManager(t *testing.T) {
+	t.Parallel()
+
+	_, err := ProvideManager(nil, nil, Options{})
+	require.Error(t, err)
```
Comment (kostola) [inline cmd/package-operator-manager/components/components_test.go:56]: What error are we expecting here?

File: cmd/package-operator-manager/components/options_test.go [hunk 4]
```diff
@@ -67,3 +67,15 @@ func TestProvideOptions(t *testing.T) {
 		},
 	}, opts)
 }
+
+func TestEnvToInt(t *testing.T) {
```
Comment (kostola) [inline cmd/package-operator-manager/components/options_test.go:71]: I think there should be two more cases
1. an existing env variable successfully converted to int
2. an existing env variable containing a random string that triggers parse error

File: cmd/package-operator-manager/bootstrap/apihelper_test.go (hunk 1 above)
Comment (Ankit152) [reply cmd/package-operator-manager/bootstrap/apihelper_test.go:287]: Here as the name suggests, we are testing for `isPKODeploymentAvailable` which is missing. The snippet you are referring is for `isPKOAvailable`. 🙂

File: cmd/package-operator-manager/components/components_test.go (hunk 2 above)
Comment (Ankit152) [reply cmd/package-operator-manager/components/components_test.go:42]: These errors are left over the way other tests in file has been left.

File: cmd/package-operator-manager/components/components_test.go (hunk 3 above)
Comment (Ankit152) [reply cmd/package-operator-manager/components/components_test.go:56]: same as above.

File: cmd/package-operator-manager/components/options_test.go (hunk 4 above)
Comment (Ankit152) [reply cmd/package-operator-manager/components/options_test.go:76]: Added the test cases.

File: cmd/package-operator-manager/bootstrap/apihelper_test.go (hunk 1 above)
Comment (kostola) [reply cmd/package-operator-manager/bootstrap/apihelper_test.go:287]: `isPKOAvailable` is internally calling both `isPKOClusterPackageAvailable` and `isPKODeploymentAvailable`.

I think these tests could be rewritten by just calling `isPKOAvailable` including all the possible test cases **only once** (the first two test cases in `TestIsPKOAvailable` and `TestIsPKODeploymentAvailable` are _exactly_ the same).

The coverage would be the same but the readability would drastically improve.

Comment (Ankit152) [reply cmd/package-operator-manager/bootstrap/apihelper_test.go]: Done.

Comment (kostola) [reply cmd/package-operator-manager/bootstrap/apihelper_test.go]: ok looks better now

**Review by kostola — APPROVED** [review]
lgtm 👍

**Pull request merged**